onnxruntime/test/framework/tunable_op_test.cc

```diff
@@ -0,0 +1,466 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+#ifdef _WIN32
+#include <windows.h>
+#endif
+#include <chrono>
+
+#include "core/common/common.h"
+#include "core/framework/tunable.h"
+
+#ifdef _WIN32
+#pragma comment(lib, "WinMM.lib")
+#endif
+
+#ifdef ORT_NO_RTTI
+#define GTEST_SKIP_IF_NO_RTTI() GTEST_SKIP() << "TunableOp needs RTTI to work correctly";
+#else
+#define GTEST_SKIP_IF_NO_RTTI()
+#endif
+
+using namespace std::chrono_literals;
+
+namespace onnxruntime {
+namespace test {
+namespace {
+
+// test on CPU and it does not use stream
+using StreamT = void*;
+
+class Timer : public ::onnxruntime::tunable::Timer<StreamT> {
+ public:
+  using TimerBase = ::onnxruntime::tunable::Timer<StreamT>;
+
+  explicit Timer(StreamT stream) : TimerBase{stream} {}
+  ~Timer() = default;
+
+  void Start() override {
+    start_ = std::chrono::high_resolution_clock::now();
+  }
+
+  void End() override {
+    end_ = std::chrono::high_resolution_clock::now();
+  }
+
+  float Duration() override {
+    return std::chrono::duration_cast<std::chrono::duration<float, std::micro>>(end_ - start_).count();
```

Comment:
Elsewhere, the `Duration` method on timers returns milliseconds, not microseconds. Suggest that this be part of the contract of the `Duration` method.

Reply:
Good catch, unified.

Reply:
Fixed